File: java/allergies/src/main/java/Allergies.java
import java.util.ArrayList;
import java.util.List;

class Allergies {

    private final List<Allergen> allergies = new ArrayList<>();

    Allergies(int score) {
        for (var allergen : Allergen.values()) {
            if ((score & allergen.getScore()) != 0) {
                allergies.add(allergen);
            }
        }
    }

    boolean isAllergicTo(Allergen allergen) {
        return allergies.contains(allergen);
    }

    List<Allergen> getList() {
        return allergies;
    }
}

File: kotlin/allergies/src/main/kotlin/Allergies.kt
class Allergies(score: Int) {

    val allergies = mutableListOf<Allergen>()

    init {
        Allergen.values()
            .forEach { allergen ->
                if (score and allergen.score != 0) {
                    allergies.add(allergen)
                }
            }
    }

    fun getList(): List<Allergen> = allergies.toList()

    fun isAllergicTo(allergen: Allergen): Boolean = allergies.contains(allergen)
}
